x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/utils/validate.ts

```diff
@@ -33,10 +36,10 @@ export const transformValidate = (
 export const transformValidateBulkError = (
   ruleId: string,
   rule: PartialRule<RuleParams>
-): RuleResponse | BulkError => {
+): RuleResponse | LegacyRuleResponse | BulkError => {
   if (isAlertType(rule)) {
     const transformed = internalRuleToAPIResponse(rule);
-    const [validated, errors] = validateNonExact(transformed, RuleResponse);
+    const [validated, errors] = validateNonExact(transformed, LegacyRuleResponse);
```

Comment:
Similar to the issue with `validate` but when using `validateNonExact` with `LegacyRuleResponse` it won't return an error if `uuid` is found in the rule actions, instead it will silently strip `uuid` out instead because `LegacyRuleResponse` doesn't expect actions to have `uuid`. So `RuleResponse` objects will be missing the `uuid` field in the response, even though they should be returned. If we tried to use `RuleResponse` for validation here instead, then Legacy rules would fail.

I think we can remove the validation step on these transformations and avoid the problem entirely.

Reply:
OK i removed validation from those endpoints, thanks.